lib/core/src/display/shape/text/text_field.rs

```diff
@@ -27,6 +27,25 @@ use crate::display::world::World;
 use nalgebra::Vector2;
 use nalgebra::Vector3;
 use nalgebra::Vector4;
+use std::ops::Range;
+
+
+
+// ===============================
+// === TextChangedNotification ===
+// ===============================
+
+/// A notification about text change.
+///
+/// In essence, it's `TextChange` with some additional useful information.
+#[derive(Clone,Debug,Shrinkwrap)]
+pub struct TextChangedNotification {
+    /// A change which has occurred.
+    #[shrinkwrap(main_field)]
+    pub change : TextChange,
+    /// The replaced range as char positions from document begin, instead of row:column pairs.
```

Comment:
I don't agree. Because this struct is not a _class_ in terms of OOP - is just a data structure. There are no functions which are requiring this invariant so far. If one receives such structure as notification and change one field - it's only his responsibility